tools/pull_request_hooks/autoLabel.js

```diff
@@ -0,0 +1,191 @@
+import * as autoLabelConfig from './autoLabelConfig.js';
+
+function keyword_to_cl_label() {
+	const keyword_to_cl_label = {};
+	for (let label in autoLabelConfig.changelog_labels) {
+		for (let keyword of autoLabelConfig.changelog_labels[label].keywords) {
+			keyword_to_cl_label[keyword] = label;
+		}
+	}
+	return keyword_to_cl_label;
+}
+
+// Checks the body (primarily the changelog) for labels to add
+function check_body_for_labels(body) {
+	const labels_to_add = [];
+
+	// if the body contains a github "fixes #1234" line, add the Fix tag
+	const fix_regex = new RegExp(`(fix[des]*|resolve[sd]*)\s*#\d+`, 'gmi');
+	if (fix_regex.test(body)) {
+		labels_to_add.push('Fix');
+	}
+
+	const keywords = keyword_to_cl_label();
+
+	let found_cl = false;
+	for (let line of body.split('\n')) {
+		if(line.startsWith(':cl:')) {
+			found_cl = true;
+			continue;
+		} else if(line.startsWith('/:cl:')) {
+			break;
+		} else if(!found_cl) {
+			continue;
+		}
+		// see if the first segment of the line is one of the keywords
+		const found_label = keywords[line.split(':')[0]?.toLowerCase()];
+		if (found_label) {
+			// don't add a billion tags if they forgot to clear all the default ones
+			const line_text = line.split(':')[1].trim();
+			const cl_label = autoLabelConfig.changelog_labels[found_label];
+			if (line_text !== cl_label.default_text && line_text !== cl_label.alt_default_text) {
+				labels_to_add.push(found_label);
+			}
+		}
+	}
+	return labels_to_add;
+}
+
+// Checks the title for labels to add
+function check_title_for_labels(title) {
+	const labels_to_add = [];
+	const title_lower = title.toLowerCase();
+	for (let label in autoLabelConfig.title_labels) {
+		let found = false;
+		for (let keyword of autoLabelConfig.title_labels[label].keywords) {
+			if (title_lower.includes(keyword)) {
+				found = true;
+				break;
+			}
+		}
+		if (found) {
+			labels_to_add.push(label);
+		}
+	}
+	return labels_to_add;
+}
+
+function check_diff_line_for_element(diff, element) {
+	const tag_re = new RegExp(`diff --git a/${element}/`);
```

Comment:
```suggestion
	const tag_re = new RegExp(`^diff --git a/${element}/`);
```
very far edge case but yeah